tap_pendo/streams.py

```diff
@@ -435,14 +435,13 @@ def send_request_get_results(self, req):
 
         resp.raise_for_status() # Check for requests status and raise exception in failure
 
-        # Separated yielding of records into a new function and called that here
-        # so that any exception raised from the session.send can be thrown from here and
-        # handled properly using backoff on request function.
-        return self.send_records(resp)
+        # get records from raw response
+        records = ijson.items(resp.raw, 'results.item')
```

Comment:
The error was raised from `ijson.items(resp.raw, 'results.item')` but as `send_records()` function was yielding data so backoff was not working properly. So, moved `ijson.items(resp.raw, 'results.item')` to `send_request_get_results()` function above which is not yielding so any error from that line will be handled properly now. The backoff for ProtocolError, ConnectionResetError was already there but it was not working properly due to the yielding function.